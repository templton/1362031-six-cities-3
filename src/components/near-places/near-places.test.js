import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NearPlaces from "./near-places";

const mockStore = configureStore([]);

const defaultCity = [52.38333, 4.9];

const neighbourhoodPlaces = [
  {
    id: 12,
    title: `Комната №12`,
    isPremium: false,
    image: `https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg`,
    costPerNignt: 150,
    type: `room`,
    raiting: 3.9,
    cord: [52.3909553943508, 4.90309666406198]
  },
  {
    id: 13,
    title: `Комната №13`,
    isPremium: false,
    image: `img/room.jpg`,
    costPerNignt: 240,
    type: `apartment`,
    raiting: 3.5,
    cord: [52.3909553943508, 4.91309666406198]
  },
  {
    id: 14,
    title: `Комната №14`,
    isPremium: false,
    image: `img/room.jpg`,
    costPerNignt: 370,
    type: `house`,
    raiting: 2.1,
    cord: [52.3909553943508, 4.92309666406198]
  }
];

const city = {
  id: 17,
  name: `Амстердам`,
  cord: [52.38333, 4.9]
};

const cities = [
  {
    id: 17,
    name: `Амстердам`,
    cord: [52.38333, 4.9]
  },
  {
    id: 15,
    name: `Барнаул`,
    cord: [53.346785, 83.776856]
  },
  {
    id: 16,
    name: `Новосибирск`,
    cord: []
  }
];

const store = mockStore({
  citiesList: cities,
  city
});

describe(`Render near places`, () => {
  it(`Simple render`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <NearPlaces
              neighbourhoodPlaces={neighbourhoodPlaces}
              onClickCardTitle={()=>{}}
              cityCord={defaultCity}/>
          </Provider>, {
            createNodeMock: () => {
              return document.createElement(`div`);
            }
          }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
