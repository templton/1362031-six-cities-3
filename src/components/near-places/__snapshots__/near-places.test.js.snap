// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render near places Simple render 1`] = `
Array [
  <section
    className="property"
  >
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </section>
  </section>,
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  150
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Комната №12
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/room.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  240
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Комната №13
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="near-places__card place-card"
          onMouseEnter={[Function]}
        >
          <div
            className="place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/room.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  370
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button button"
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
              onClick={[Function]}
            >
              <a
                href="#"
              >
                Комната №14
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              house
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>,
]
`;
